src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -179,14 +182,23 @@ CATCH_RETURN()
                 _fontInUse,
                 &isTextSimple,
                 &uiLengthRead,
-                &_glyphIndices.at(pos));
+                &runGlyphIndices.at(pos));
 
             RETURN_IF_FAILED(hr);
             _SetCurrentRun(pos);
             _SplitCurrentRun(pos);
+
+            if (isTextSimple)
+            {
+                // Only copy the indices when the run is simple.
+                // Otherwise the glyph indices from GetTextComplexity are not meaningful.

```

Comment:
Even with the fix, the `_glyphIndices` for simple runs will have *wrong start* index, if there's N-M runs previous to them. For example a simple line like this:

```text
😀
```

There's two runs. First is complex, the second is simple. 

| Text | Length | Glyph Count |
| -- | -- | -- |
| 😀| 2 | 1 |
| (A lot of spaces | 118 | 118 |

The correct `_glyphIndices` is this:

| Index | Value |
| -- | -- |
| 0  | 11685 |
| 1  | 3 |
| 2  | 3 |
| 3 | 3 | 
| ... | ... |

But here we have:

| Index | Value |
| -- | -- |
| 0  | 11685 |
| 1  | **0** |
| 2  | 3 |
| 3 | 3 | 
| ... | ... |

Notice there's an incorrect 0 indice here, because when we do `GetTextComplexity` on the second run, we have no idea how many glyphs the previous run takes up, and assume the number is equal to the textLength (2) and fill up the `_glyphIndices` from there.